### ISSUE TITLE
CLI stucks if no devices found

### ISSUE THREAD

**Issue by tagantroy** [MarathonLabs/marathon#143]
I reproduced it on sample-app
```
Listening for transport dt_socket at address: 1044
14:02:55.487 [main] INFO com.malinskiy.marathon.cli.ApplicationView - Starting marathon
14:02:55.737 [main] INFO com.malinskiy.marathon.cli.config.ConfigFactory - Checking Marathonfile config
Exception in thread "main" com.malinskiy.marathon.exceptions.NoDevicesException: No devices found.
	at com.malinskiy.marathon.android.AndroidDeviceProvider.initialize(AndroidDeviceProvider.kt:144)
	at com.malinskiy.marathon.Marathon.loadDeviceProvider(Marathon.kt:62)
	at com.malinskiy.marathon.Marathon.runAsync(Marathon.kt:84)
	at com.malinskiy.marathon.Marathon$run$1.invokeSuspend(Marathon.kt:76)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:236)
	at kotlinx.coroutines.EventLoopBase.processNextEvent(EventLoop.kt:123)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:69)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:45)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:35)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at com.malinskiy.marathon.Marathon.run(Marathon.kt:75)
	at com.malinskiy.marathon.cli.ApplicationViewKt$main$1.invoke(ApplicationView.kt:42)
	at com.malinskiy.marathon.cli.ApplicationViewKt$main$1.invoke(ApplicationView.kt)
	at com.xenomachina.argparser.SystemExitExceptionKt.mainBody(SystemExitException.kt:74)
	at com.xenomachina.argparser.SystemExitExceptionKt.mainBody$default(SystemExitException.kt:72)
	at com.malinskiy.marathon.cli.ApplicationViewKt.main(ApplicationView.kt:24)
```

**Comment by DSamaryan** [comment]
I found the same issue starting through the gradle plugin. That problem might be avoided if the device is reconnected while AndroidDeviceProvider is waiting for any device. It makes AndroidDebugBridge send the events which will be handled in AndroidDebugBridge.IDeviceChangeListener

**Comment by tagantroy** [comment]
Fixed #162
